### PR TITLE
Fixed 'Invalid column COUNT(*) as count' on CameraRoll.getAlbums()

### PR DESCRIPTION
**Pull request by dokkis**
Fixed 'Invalid column COUNT(*) as count' on CameraRoll.getAlbums() due to GROUP BY support removed in Android Q.

# Summary

CameraRoll.getAlbums() has 2 issues:

1. Exception thrown: java.lang.IllegalArgumentException: Invalid column COUNT(*) as count with API Level 29.
2. Exception get swallowed and doesn't return the promise here https://github.com/react-native-community/react-native-cameraroll/blob/master/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java#L430 so the function called CameraRoll.getAlbums() never get into the `then` or the `catch` part from client perspective
For the issue #1, it looks like GROUP BY support is removed from Android Q, that's why is not working.

Related to: https://github.com/react-native-community/react-native-cameraroll/issues/174

### What's required for testing (prerequisites)?
Device/Emulator with API Level 29.

### What are the steps to reproduce (after prerequisites)?
Use the getAlbums().

```
CameraRoll.getAlbums()
.then( (res) => console.log(res))
.catch( (err) => console.error(err))
```

This breaks in API Level 29 and the current PR support API Level 29 and below as well, it's using the same methodology without using `GROUP BY` that got removed in Android Q and having the group by logic by scrolling over all media using the cursor and count them.

I verified that the output of the method is the same by using `All`, `Photos` or `Videos` as assetType input compared to the old implementation. Both are ordered and both counts the same number of elements in the 3 above scenario in my device that has around 10k of photos.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ❌     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [n/a] I added the documentation in `README.md`
- [n/a] I updated the typed files (TS and Flow)
- [n/a] I added a sample use of the API in the example project (`example/App.js`)

It's not a new functionality but a bug fix so no update on README or interface is required.

### CODE REVIEW

**Comment by dokkis** [comment]
@bartolkaruza can you kindly look into my PR? the release build also failed, but I am not sure if it's caused by my code, in the logs I see `AAPT2 aapt2-3.2.1-4818971-linux Daemon #0: Unexpectedly exit.`. I cannot retry the workflow to see if succeed. Thanks!

File: android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java [hunk 1]
```diff
@@ -413,22 +410,34 @@ public void getAlbums(final ReadableMap params, final Promise promise) {
         WritableArray response = new WritableNativeArray();
         try {
           if (media.moveToFirst()) {
+            Map<String, Integer> albums = new TreeMap<>();
```
Comment (r0b0t3d) [inline android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java:413]: What is the purpose of `TreeMap` here? If you just want to order album name, I think we do not need to do that. Basically, lib just return the album list, user may want to order the list by themself (by name or by count).

File: android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java [hunk 2]
```diff
@@ -413,22 +410,34 @@ public void getAlbums(final ReadableMap params, final Promise promise) {
...
+            }
           }
         } finally {
           media.close();
           promise.resolve(response);
         }
       }
-    } catch (Exception e) {}
-
+    } catch (Exception e) {
```
Comment (r0b0t3d) [inline android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java:438]: Could you consider to merge 2 try catch block into 1? the original implementation got problem here.

File: android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java (hunk 1 above)
Comment (dokkis) [reply android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java:413]: that was mainly for ordering, I can change to `HashMap` if we don't care about order and eventually clients can order on their and if they need to. Should I?

File: android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java (hunk 2 above)
Comment (dokkis) [reply android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java:438]: the best would be using the try-with-resources construct for the Cursor but is supported from API Level 19, If you do one main try you would need to have the `media` cursor reference outside of the try and check the nullability on the finally. I think I would rather keep as it is.

File: android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java (hunk 1 above)
Comment (r0b0t3d) [reply android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java:413]: It would be great!

**Comment by bartolkaruza** [comment]
@dokkis the release build was just a glitch, I reran the workflow and it works fine. Thank you for the contribution!

**Pull request merged**